content-services/content-sources-backend
# Fixes 1469: delete snapshots when deleting repo config

## Summary
Starts a task to delete snapshots and pulp references when deleting a repository config. If creating a snapshot is in progress when the delete request is made, the API will reject the request to delete the repository config.

## Testing steps
1. Create a repository
2. Wait for the sync task to complete
3. Make note of the pulp hrefs created for the snapshot (remote, repository, repository version, distribution).
4. Delete the repository configuration
5. Wait for the delete task to complete, then check the earlier pulp hrefs to see that they were deleted.
6. Create a repository again
7. Delete the repository before the sync task completes, your request to delete the repository should be rejected.

Note that this doesn't handle the delete snapshot task exiting early. If the server closes during this task, the snapshot/pulp objects may or may not be deleted.

Todo
